src/Invoice/Inv/Trait/Calendar.php

```
@@ -0,0 +1,176 @@
+        $months = [];
+        for ($offset = -self::MONTHS_BEFORE; $offset <= self::MONTHS_AFTER; $offset++) {
+            $months[] = $this->calendarModify($target, $offset . ' months');
+        }
+        return $months;
+    }
+
+    /**
+     * @return array<string, array<int, int>> [day 'Y-m-d' => [categorySecondaryId => invoice count]]
+     */
+    private function calendarBucketInvoicesByDay(
+        IR $iR,
```

PHPMD (2.15.0)

[warning] Avoid variables with short names like $iR. Configured minimum length is 3. (undefined)

(ShortVariable)